api/lock.go

```diff
@@ -165,25 +165,27 @@ WAIT:
 	if pair != nil && pair.Flags != LockFlagValue {
 		return nil, ErrLockConflict
 	}
-	if pair != nil && pair.Session != "" {
+	if pair != nil && pair.Session != l.opts.Session {
```

Comment:
Much simpler is above this line just add:
```
if pair != nil && pair.Session = l.lockSession {
  goto HELD
}
```
Then add the `HELD` label above `leaderCh := make(chan struct{})`